components/insight/SRC/org/openmicroscopy/shoola/agents/fsimporter/view/FailedImportDialog.java

```diff
@@ -0,0 +1,168 @@
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *  
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *------------------------------------------------------------------------------
+ */
+package org.openmicroscopy.shoola.agents.fsimporter.view;
+
+import java.awt.BorderLayout;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import javax.swing.BorderFactory;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+
+import org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI;
+import org.openmicroscopy.shoola.util.file.ImportErrorObject;
+import org.openmicroscopy.shoola.util.ui.UIUtilities;
+
+/**
+ * A dialog for displaying the error logs for files which failed to import.
+ * 
+ * @author Domink Lindner &nbsp;&nbsp;&nbsp;&nbsp; <a
+ *         href="mailto:d.lindner@dundee.ac.uk">d.lindner@dundee.ac.uk</a>
+ */
+public class FailedImportDialog extends JDialog {
+
+    /** Holds the error logs (key: file name) */
+    private Map<String, String> errors = new HashMap<String, String>();
+
+    /**
+     * Create new instance
+     * 
+     * @param owner
+     *            The dialog owner
+     * @param components
+     *            The failed imports
+     */
+    public FailedImportDialog(JFrame owner,
+            Collection<FileImportComponentI> components) {
+        super(owner);
+
+        for (FileImportComponentI c : components) {
+            ImportErrorObject obj = c.getImportErrorObject();
+            errors.put(UIUtilities.formatPartialName(obj.getFile()
+                    .getAbsolutePath(), 60), UIUtilities.printErrorText(obj
+                    .getException()));
+        }
+
+        buildUI();
+    }
+
+    private void buildUI() {
+        setTitle("Failed imports");
+
+        JPanel con = new JPanel();
+        con.setBorder(BorderFactory.createEmptyBorder(4, 4, 4, 4));
+        con.setLayout(new GridBagLayout());
+
+        GridBagConstraints c = new GridBagConstraints();
+        c.ipadx = 2;
+        c.ipady = 2;
+
+        c.gridx = 0;
+        c.gridy = 0;
+        c.fill = GridBagConstraints.NONE;
+        con.add(new JLabel("File:"), c);
+
+        String[] tmp = null;
+        if (errors.keySet().size() < 50) {
```

Comment:
Maybe replace the 50-related stuff into something using a `final int WHATEVER = 50`?